Expecto/Expecto.fs

```diff
@@ -1151,14 +1170,23 @@ module Impl =
           findBase t.BaseType
       findBase t
     baseType.IsGenericType && baseType.GetGenericTypeDefinition() = typedefof<FSharpFunc<unit, unit>>
+#endif
 
   let getFuncTypeToUse (testFunc:unit->unit) (asm:Assembly) =
     let t = testFunc.GetType()
+#if RESHAPED_REFLECTION
+    if t.GetTypeInfo().Assembly.FullName = asm.FullName then
+#else
     if t.Assembly.FullName = asm.FullName then
+#endif
       t
     else
       let nestedFunc =
+#if RESHAPED_REFLECTION
```

Comment:
There shouldn't be a need for this, unless there is some performance issue to call the GetTypeInfo wrappers?